tests/python/gaia-ui-tests/gaiatest/apps/settings/regions/bluetooth.py

```diff
@@ -35,15 +35,21 @@ def device_name(self):
     def enable_bluetooth(self):
         self.marionette.find_element(*self._bluetooth_label_locator).tap()
         checkbox = self.marionette.find_element(*self._bluetooth_checkbox_locator)
-        Wait(self.marionette).until(expected.element_selected(checkbox))
+        Wait(self.marionette).until(lambda m: self.is_custom_element_checked(checkbox))
         rename_device = self.marionette.find_element(*self._rename_my_device_button_locator)
         Wait(self.marionette).until(expected.element_enabled(rename_device))
 
+    def disable_bluetooth(self):
+        self.marionette.find_element(*self._bluetooth_label_locator).tap()
+        checkbox = self.marionette.find_element(*self._bluetooth_checkbox_locator)
+        Wait(self.marionette).until(lambda m: not self.is_custom_element_checked(checkbox))
+        rename_device = self.marionette.find_element(*self._rename_my_device_button_locator)
+        Wait(self.marionette).until(expected.element_not_displayed(rename_device))
+
     def enable_visible_to_all(self):
-        # Bluetooth state is stored outside the profile bug 969310
         self.marionette.find_element(*self._visible_to_all_label_locator).tap()
```

Comment:
Same here, can't you use checkbox.tap() here?
Then, you can get rid of teh _visible_to_all_label_locator locator.